flybywiresim/a32nx
# fix(engine): fuel flow and fuel consumption model

Fixes #4656

## Summary of Changes

Fixed a couple of issues with fuel flow and fuel consumption model.
- Fuel flow: instead of staying at 0 at idle after start-up (does not happen on runway), it now ramps up from 0 to its idle figure
- Fuel consumption: fixed the infinite fuel issue and random fuel increases/ rebounds

Disclaimer: 
Fuel flow is ramping now during start-up but not as in real life (work in progress). 
Regarding fuel consumption, please disregard fuel feeding between tanks as this is being worked on by the team. Also, there are reports concerning auxiliary tanks not filling inner tanks when start fuel is below sensor threshold. This only happens when starting on runway (does not happen if starting C&D). This issue will also have to be taken care of with the new fuel model being designed. 

## Screenshots (if necessary)



## References





## Additional context



Discord username (if different from GitHub): TazX [Z+2]#0405

## Testing instructions

1. For fuel flow:
- Start C&D
- Proceed to Engine Start
- Watch engine spool up to idle and check fuel flow starts at 0 and increments up to around 300 Kg/h (700 lbs/h)

2. For fuel consumption:
- During flight and/ or on runway:
    - Check fuel burn corresponds to fuel flow figures
    - Check when tanks are empty, fuel is 0 and engines turn off


## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page
